### ISSUE TITLE
(re)Introduce the CSS-WG to Warichu

### ISSUE THREAD

**Issue by frivoal** [w3c/csswg-drafts#9520]
This is an issue to remind myself to write a lightning talk for the next CSSWG F2F to give an intro to Warichu layout.

**Comment by xfq** [comment]
Although I don't have the time to write a lightning talk myself, I can provide some relevant information.

For Chinese:

It is common in commentaries on classical books, books about Chinese herbology, scripts (screenplays), reference books, etc.

[clreq](https://www.w3.org/TR/clreq/) does not currently have information about warichu, but we plan to add some text, especially regarding modern usage. (Ancient usage is different.)

AFAIK there are two forms, the single-line form (see horizontal example [1](https://user-images.githubusercontent.com/8192599/86930139-9f854d00-c171-11ea-835e-93721ad4dd17.png) [2](https://user-images.githubusercontent.com/2863444/273788844-ce4569f2-f940-4e4a-b5cf-bf69b4ef6c4e.jpg) and [vertical example](https://user-images.githubusercontent.com/8192599/86930981-8c26b180-c172-11ea-8e53-e193f32afd7e.png)) and the double-line form. In modern usage, the single-line form is more popular than the double-line form.

The font size of warichu must be smaller than the font size of the body text. Punctuation marks in warichu should treated the same as in the main text, including the line breaking rules.

Warichu can be set solid with the body text or extra spacing can be added, depending on the house rule. Sometimes parentheses can be added at the range start and range end. The parentheses are also subjected to inline [punctuation width adjustment](https://www.w3.org/TR/clreq/#punctuation_width_adjustment). The height of the parentheses for the double-line form is usually equal to the height of the two lines of text, and is usually half-width.

Single-line warichu should be aligned with the bottom of the text when arranged horizontally, and aligned with the right side of the text when arranged vertically. Regardless of whether they are arranged horizontally or vertically, the two lines of double-line warichu should be center-aligned with the main text. That is to say, although the total height of the two-line warichu will be greater than the main text, the protruding parts should be evenly distributed to the two sides. Regardless of whether there is warichu or not, the [line gap](https://www.w3.org/TR/clreq/#endef_lineGap) of the body text should remain consistent (except that the line gap will become smaller where there are double-line warichu).

The line gap between the two lines inside the double-line warichu is usually 0. The two lines in a double-lined warichu should be as close to the same length as possible. When the warichu does not need to span the line break position of the main text, the annotation should be broken into two lines as evenly as possible at a place where it can be broken, and the line breaking rules should be consistent with the line breaking rules of the main text. When two lines are of different lengths, the second line should be shorter than the first.

Depending on where the warichu is inserted and the length of the warichu itself, the annotation may need to span a line break in the main text. The processing at this time is still under discussion in the clreq group. I'll let you know the results of the discussions.

Also, warichu characters are often taller and thinner than the body text (see [an example](https://user-images.githubusercontent.com/20491276/90466362-30bdeb00-e144-11ea-8768-6cf58652f878.jpg)).

-----

For Japanese, see [JLReq](https://www.w3.org/TR/jlreq/#inline_cutting_note) for more information.

**Comment by kojiishi** [comment]
Very interesting information @xfq, thank you.

From what you wrote, I can find a couple of differences between Chinese and Japanese:
* Japanese also has a pattern to set notes in smaller size with parentheses, but we don't call them "Wari-Chu". The "Wari" means "divide/split", the "Chu" means "note". One line case for Japanese is just "Chu" (note,) without being split.
* The smaller inline note is aligned center in Japanese, both in horizontal and in vertical.

There may be more, but these are my first findings. /cc @Kaleidea @himorin

**Comment by xfq** [comment]
> Japanese also has a pattern to set notes in smaller size with parentheses, but we don't call them "Wari-Chu". The "Wari" means "divide/split", the "Chu" means "note". One line case for Japanese is just "Chu" (note,) without being split.

Indeed. In the clreq group we're discussing a new English term that covers both meanings.

The term "Chu" is so broad that it could theoretically encompass things like 後注, 頭注, 脚注, 傍注, and 行間注, and we would like to avoid that term if possible.

**Comment by frivoal** [comment]
https://florian.rivoal.net/talks/warichu/

**Comment by xfq** [comment]
> https://florian.rivoal.net/talks/warichu/

Thank you, Florian!

-----

In the slides:

> Also used (more rarely) in Chinese

Any specific data or sources for this? I don't have any data, but I doubt that's true (about the "more rarely" part).

**Comment by frivoal** [comment]
>> Also used (more rarely) in Chinese
> Any specific data or sources for this? I don't have any data, but I doubt that's true (about the "more rarely" part).

Wikipedia claims it's more rare (https://en.wikipedia.org/wiki/Warich%C5%AB#Modern_use) but has no source about that. I'll drop that claim.

**Comment by xfq** [comment]
> dholbert: any scenarios where this might apply recursively?

FYI, there are real examples of warichū embedded inside a warichū, like [this image](https://private-user-images.githubusercontent.com/2863444/285087418-6dc8b804-8e18-4283-89f1-03c5e961896b.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4Njk3NDMsIm5iZiI6MTcwNzg2OTQ0MywicGF0aCI6Ii8yODYzNDQ0LzI4NTA4NzQxOC02ZGM4YjgwNC04ZTE4LTQyODMtODlmMS0wM2M1ZTk2MTg5NmIuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMTRUMDAxMDQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTM3MWZkODQyMzkzY2UxNTNjOGUyNTJlZTVhYmNjZjU1NTNmN2Q1ZmY3Yzk4YjM1YzhkZDg5OWI5ZjBlMTRlYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.zg8JeZd1PPxV-9LauJy67SCfwbCaTeluCZsjFfdHxM8) (the four characters below the red box on the left).

**Comment by tabatkins** [comment]
@xfq that image isn't publicly available

**Comment by xfq** [comment]
> @xfq that image isn't publicly available

Sorry, I've updated the URL.
